xarray/conventions.py

```diff
@@ -350,6 +350,25 @@ def stackable(dim):
         drop_variables = []
     drop_variables = set(drop_variables)
 
+    # Time bounds coordinates might miss the decoding attributes
+    # https://github.com/pydata/xarray/issues/2565
+    if decode_times:
+        # We list all time variables with bounds
+        to_update = dict()
+        for v in variables.values():
+            attrs = v.attrs
+            if ('units' in attrs and 'since' in attrs['units'] and
```

Comment:
It might be cleaner to write something like:
```python
has_date_units = 'units' in attrs and 'since' in attrs['units']
if has_date_units and 'bounds' in attrs:
```